timestamp-impl/src/test/java/com/palantir/timestamp/TimestampRangesTest.java

```diff
@@ -19,95 +19,135 @@
 import static org.assertj.core.api.Assertions.assertThat;
 import static org.assertj.core.api.Assertions.assertThatThrownBy;
 
+import java.util.OptionalLong;
+
 import org.junit.Test;
+import org.junit.experimental.theories.DataPoints;
+import org.junit.experimental.theories.Theories;
+import org.junit.experimental.theories.Theory;
+import org.junit.runner.RunWith;
 
+@RunWith(Theories.class)
 public class TimestampRangesTest {
     private static final long SEVENTY_THREE = 73L;
     private static final long EIGHTY_TWO = 82L;
 
     private static final TimestampRange SEVENTY_THREE_TO_EIGHTY_TWO
             = TimestampRange.createInclusiveRange(SEVENTY_THREE, EIGHTY_TWO);
 
-    @Test
-    public void canGetTimestampFromRangeIfItIsTheLowerBound() {
-        assertThat(TimestampRanges.getTimestampMatchingModulus(SEVENTY_THREE_TO_EIGHTY_TWO, 3, 10))
+    @DataPoints
+    public static SearchOption[] searchOptions = SearchOption.values();
+
+    @Theory
+    public void canGetTimestampFromRangeIfItIsTheLowerBound(SearchOption searchOption) {
+        assertThat(searchOption.timestampMatchingModulus(SEVENTY_THREE_TO_EIGHTY_TWO, 3, 10))
                 .isPresent()
                 .hasValue(SEVENTY_THREE);
     }
 
-    @Test
-    public void canGetTimestampFromRangeIfItIsTheUpperBound() {
-        assertThat(TimestampRanges.getTimestampMatchingModulus(SEVENTY_THREE_TO_EIGHTY_TWO, 2, 10))
+    @Theory
```

Comment:
fwiw, I've found that theories make it very hard to actually debug test failures. If you fail on only one case, it's tricky to figure out the exact failure mode.

Reply:
removing theories, as we don't need to use two separate search options anymore